examples/documentation/tests/bindings/test_documentation.py

```diff
@@ -0,0 +1,19 @@
+import unittest
+import documentation
+
+class TestHello(unittest.TestCase):
+    def test_hello(self):
+        self.assertEqual(documentation.hello(documentation.Pet("Tom")),
+                         "Hello Tom!", "Should be `Hello Tom!`")
+
+class TestGetName(unittest.TestCase):
+    def test_get_name(self):
+        self.assertEqual(documentation.Person("Daniel").get_name(), "Daniel", "Should be Daniel")
+        
+class TestAdd(unittest.TestCase):
+    def test_add(self):
+        self.assertEqual(documentation.add(2, 4), 6, "Should be 6")
+
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
It seems like this is testing the functionality rather than the doc generation.

What if we made these tests work more like the trybuild tests?  We check source files that have the classes/functions/whatever, with the docstrings, but with stub code for their bodies.  Then test that the generated code has docstrings that match the expected ones.  I think that's testing this change better and also users can browse the checked in files to get a feeling of what the generated docs look like.

Reply:
I agree - as written this test doesn't seem to have any value - it's testing things with already good coverage and doesn't seem to be testing the new functionality added here.

Reply:
I have implemented a "unit" test to ensure that generated bindings contain appropriate docstrings. The test is written in a fixture. Inside the UDL file, each uniffi node (`function`, `object`, `enum`, etc..) has a different docstring attached to it. The test generates bindings from the UDL file. Then, the test does `cat|grep` to ensure that docstrings are being generated for all nodes. While this test doesn't check that docstrings are placed in the appropriate location, it does at least ensure that all nodes attempt to generate some form of their docstrings.

You can check the code here: https://github.com/NordSecurity/uniffi-rs/compare/kristupas/docstrings...NordSecurity:uniffi-rs:kristupas/docstring-templates?expand=1

Reply:
Unit test is just a part of an example - it does not test doc generation.
IMO doc comments generation should be tested separately, outside of examples (you shouldn't be confusing `uniffi-rs` users with `uniffi-rs` internal functionality testing inside examples) - how to do it is still TBD.